crypto/src/main/kotlin/web5/sdk/crypto/Ed25519.kt

```diff
@@ -43,67 +42,80 @@ public object Ed25519 : KeyGenerator, Signer {
    * Generates a private key utilizing the Ed25519 algorithm.
    *
    * @param options (Optional) Additional options to control the key generation process.
-   * @return The generated private key in JWK format.
+   * @return The generated private key in Jwk format.
    */
-  override fun generatePrivateKey(options: KeyGenOptions?): JWK {
-    return OctetKeyPairGenerator(com.nimbusds.jose.jwk.Curve.Ed25519)
-      .algorithm(JWSAlgorithm.EdDSA)
-      .keyIDFromThumbprint(true)
-      .keyUse(KeyUse.SIGNATURE)
-      .generate()
-      .toOctetKeyPair()
+  override fun generatePrivateKey(options: KeyGenOptions?): Jwk {
+    // TODO use tink to generate private key https://github.com/TBD54566975/web5-kt/issues/273
+    val privateKey = OctetKeyPairGenerator(Curve.Ed25519).generate()
+
+    val jwk = Jwk.Builder("OKP", privateKey.curve.name)
+      .privateKey(privateKey.d.toString())
+      .x(privateKey.x.toString())
+      .build()
+
+    return jwk
+
   }
 
   /**
    * Derives the public key corresponding to a given private key.
    *
-   * @param privateKey The private key in JWK format.
-   * @return The corresponding public key in JWK format.
+   * @param privateKey The private key in Jwk format.
+   * @return The corresponding public key in Jwk format.
    */
-  override fun computePublicKey(privateKey: JWK): JWK {
-    require(privateKey is OctetKeyPair) { "private key must be an Octet Key Pair (kty: OKP)" }
+  override fun computePublicKey(privateKey: Jwk): Jwk {
+    validateKey(privateKey)
+
+    val jwk = Jwk.Builder(privateKey.kty, curve.name)
+      .algorithm(algorithm.name)
+      .apply {
+        privateKey.use?.let { keyUse(it) }
+        privateKey.alg?.let { algorithm(it) }
+        privateKey.x?.let { x(it) }
+      }
+      .build()
 
-    return privateKey.toOctetKeyPair().toPublicJWK()
+    return jwk
   }
 
-  override fun privateKeyToBytes(privateKey: JWK): ByteArray {
+  override fun privateKeyToBytes(privateKey: Jwk): ByteArray {
```

Comment:
it would be good to have methods that are able to represent compressed keys

Reply:
for `Ed25519` @decentralgabe ? or are you talking about `Secp256k1` or more generally `EC`?

Reply:
would like gabe to weigh in on this / edit this issue to clarify what's missing here

https://github.com/TBD54566975/web5-kt/issues/280